astroquery/esa/jwst/tests/test_jwsttap.py

```diff
@@ -0,0 +1,1088 @@
+        # The query contains decimals: force default response
+        connHandler.set_default_response(responseLaunchJob)
+        sc = SkyCoord(ra=29.0, dec=15.0, unit=(u.degree, u.degree),
+                      frame='icrs')
+        with pytest.raises(ValueError) as err:
+            tap.query_region(sc)
+            print(err.value.args[0])
+        assert "Missing required argument: 'width'" in err.value.args[0]
+
+        width = Quantity(12, u.deg)
+        height = Quantity(10, u.deg)
+
+        with pytest.raises(ValueError) as err:
+            tap.query_region(sc, width=width)
+        assert "Missing required argument: 'height'" in err.value.args[0]
+
+        # Test observation_id argument
+        with pytest.raises(ValueError) as err:
+            tap.query_region(sc, width=width, height=height, observation_id=1)
+        assert "observation_id must be string" in err.value.args[0]
+
+        # Test cal_level argument
+        with pytest.raises(ValueError) as err:
+            tap.query_region(sc, width=width, height=height, cal_level='a')
+        assert "cal_level must be either 'Top' or an "\
+               "integer" in err.value.args[0]
+
+        # Test only_public
+        with pytest.raises(ValueError) as err:
+            tap.query_region(sc, width=width, height=height, only_public='a')
+        assert "only_public must be boolean" in err.value.args[0]
+
+        # Test dataproduct_type argument
+        with pytest.raises(ValueError) as err:
+            tap.query_region(sc, width=width, height=height, prod_type=1)
+        assert "prod_type must be string" in err.value.args[0]
+
+        with pytest.raises(ValueError) as err:
+            tap.query_region(sc, width=width, height=height, prod_type='a')
+        assert "prod_type must be one of: " in err.value.args[0]
+
+        # Test instrument_name argument
+        with pytest.raises(ValueError) as err:
+            tap.query_region(sc, width=width, height=height, instrument_name=1)
+        assert "instrument_name must be string" in err.value.args[0]
+
+        with pytest.raises(ValueError) as err:
+            tap.query_region(sc, width=width, height=height,
+                             instrument_name='a')
+        assert "instrument_name must be one of: " in err.value.args[0]
+
+        # Test filter_name argument
+        with pytest.raises(ValueError) as err:
+            tap.query_region(sc, width=width, height=height, filter_name=1)
+        assert "filter_name must be string" in err.value.args[0]
+
+        # Test proposal_id argument
+        with pytest.raises(ValueError) as err:
+            tap.query_region(sc, width=width, height=height, proposal_id=123)
+        assert "proposal_id must be string" in err.value.args[0]
+
+        table = tap.query_region(sc, width=width, height=height)
+        assert len(table) == 3, \
+            "Wrong job results (num rows). Expected: %d, found %d" % \
+            (3, len(table))
```

Comment:
this is a typical case where I feel a single line would be more readable, all would fit into <100 chars. (f-stings can also be used)

Reply:
Suppose we have a small test module that looks like this:
```python
from astropy.table.table_helpers import simple_table

def test():
    table = simple_table(2)
    assert len(table) == 3
```
Running it produces the following:
```
=================================== FAILURES ===================================
_____________________________________ test _____________________________________

    def test():
        table = simple_table(2)
>       assert len(table) == 3
E       assert 2 == 3
E        +  where 2 = len(<Table length=2>\n  a      b     c  \nint64 float64 str1\n----- ------- ----\n    1     1.0    c\n    2     2.0    d)

set4/test_solutions.py:5: AssertionError
=========================== short test summary info ============================
FAILED set4/test_solutions.py::test - assert 2 == 3
============================== 1 failed in 0.10s ==============================
```
With the custom assertion error message the output is:
```
=================================== FAILURES ===================================
_____________________________________ test _____________________________________

    def test():
        table = simple_table(2)
>       assert len(table) == 3, \
                "Wrong job results (num rows). Expected: %d, found %d" % \
                (3, len(table))
E       AssertionError: Wrong job results (num rows). Expected: 3, found 2
E       assert 2 == 3
E        +  where 2 = len(<Table length=2>\n  a      b     c  \nint64 float64 str1\n----- ------- ----\n    1     1.0    c\n    2     2.0    d)

set4/test_solutions.py:5: AssertionError
=========================== short test summary info ============================
FAILED set4/test_solutions.py::test - AssertionError: Wrong job results (num ...
============================== 1 failed in 0.11s ===============================
```
I don't see much difference between the outputs. Wouldn't it be simpler not to write the custom assertion error message or is there some benefit I'm not noticing?

Reply:
I myself don't write custom assert messages, and the generic part of astroquery or other libraries I'm more involved in don't have them. But I fully accept and support if a maintainer group of a module find those useful. So here I would totally leave it to @jespinosaar and others at ESA to have their preference. My comment was really just an example that the we don't need to stick with line lengths of 80 especially when the code could be made more readable and compact using a little longer lines, yet staying within 100-120.

Reply:
Text has been reformatted